### ISSUE TITLE
"Pythonic" field naming

### ISSUE THREAD

**Issue by mo-lukecarr** [KNMI/edr-pydantic#19]
Could we use Pydantic's [field aliases](https://docs.pydantic.dev/latest/concepts/fields/#field-aliases) to use "Pythonic" field names (i.e. snake case) in the models, without losing the compatibility of validating and serialising the EDR specification?

This work might be blocked by the fact that Pydantic currently requires you to explicitly pass `by_alias=True` when dumping a model instance. There doesn't appear to be a way of automatically enabling `by_alias`, such as through a model's `ConfigDict`.

If we were to implement this change and introduce aliases, we'd probably need to have a way of defaulting to validating and serialising by alias (otherwise the library would rely on users to remember to always pass `by_alias=True`).

**Comment by PaulVanSchayck** [comment]
We recently had to enable `by_alias=` to support Parameter.`data-type`, and were a bit shocked by how bad the alias support still is in Pydantic. See our PR (https://github.com/KNMI/edr-pydantic/pull/24) and the related [Pydantic issue](https://github.com/pydantic/pydantic/issues/8379).

But even if they would improve alias support, this is most likely still a minefield to get into. While the benefit ('my code formatting is more consistent') is pretty minor. Although I realise that opinions do differ here. 

I've added a `wontfix` label, but I'm happy to be convinced otherwise.
